assets/js/base/hooks/shipping/use-shipping-rates.js

```diff
@@ -56,7 +65,47 @@ export const useShippingRates = ( addressFieldsKeys ) => {
 			) &&
 			debouncedShippingAddress.country
 		) {
-			updateShippingAddress( debouncedShippingAddress );
+			updateShippingAddress( debouncedShippingAddress ).catch(
+				( error ) => {
+					switch ( error.code ) {
+						case shippingErrorCodes.INVALID_COUNTRY:
+							addErrorNotice(
+								__(
+									'The country provided is not a valid country for this site. Please select a different country.',
+									'woo-gutenberg-products-block'
+								),
+								{
+									id: 'shipping-address-error',
+								}
+							);
+							return;
+						case shippingErrorCodes.INVALID_STATE:
+							addErrorNotice(
+								__(
+									'The provided state/county/province for the given country is not valid. Please select a different state/county/province.',
+									'woo-gutenberg-products-block'
+								),
+								{
+									id: 'shipping-address-error',
+								}
+							);
+							return;
+						case shippingErrorCodes.MISSING_COUNTRY:
+							addErrorNotice(
+								__(
+									'Country is required for calculating rates. Please select a country',
+									'woo-gutenberg-products-block'
+								),
+								{
+									id: 'shipping-address-error',
+								}
+							);
+							return;
```

Comment:
Shouldn’t the server return this data for us? and we simply pass it down?

Reply:
The server has a message suitable for _developers_ hitting the endpoint and getting an error. This is a more customer friendly message. We have to keep in mind the context of who is reading these messages: developers, merchants, or the merchant's customers?